test/rewrite_test.go

```diff
@@ -72,15 +110,43 @@ func testEdns0(t *testing.T, server string) {
 	if o == nil || len(o.Option) == 0 {
 		t.Error("Expected EDNS0 options but got none")
 	} else {
+		if len(o.Option) != 1 {
+			t.Errorf("Expected only one EDNS0 options but got %d", len(o.Option))
+		}
 		if e, ok := o.Option[0].(*dns.EDNS0_LOCAL); ok {
 			if e.Code != 0xffee {
 				t.Errorf("Expected EDNS_LOCAL code 0xffee but got %x", e.Code)
 			}
-			if !bytes.Equal(e.Data, []byte("hello-world")) {
-				t.Errorf("Expected EDNS_LOCAL data 'hello-world' but got %q", e.Data)
+			if !bytes.Equal(e.Data, []byte("initialization")) {
```

Comment:
Why did you change this test? It seems the "initialization" should be rewritten with "hello-world" by rewrite plugin

Reply:
That is the whole point of this comment here : https://github.com/coredns/coredns/issues/2484#issuecomment-456433496

With my changes:

1- client EDNS_LOCAL("Initialization") no EDNS
2- Rewrite EDNS_LOCAL("Hello-World")
3- erratic resolution --> no EDNS
4- Rewrite writing response : let the response flow .. (do not alter the response at all)
5- SizeAndDo : if response has no EDNSO, fill with queries EDNS0 else let the response flow


Question: Should step 4 consolidate the response with the EDNS of the forged query at step 2 ?

I think no, but not sure here ... it is interpretation of the RFC.
I would think that the forged query of REWRITE is ONLY to get a resolution. It should be transparent for the initial client - but I maybe wrong.

Let me know what you think.

If we need to revert to former behavior, I guess that remove the last commit will do the trick.

Reply:
> Why did you change this test? It seems the "initialization" should be rewritten with "hello-world" by rewrite plugin

I do not think so.
Right now, for unclear reason, SizeAndDo() is just copying the request ENS0s  into the response OPT.
But i think we just should copy the header OPT (to confirm the version EDNS0). But there is no reason to copy the EDNS0 elements ... (Or I do not see it).

IMO, the initial UT that is testing 'Hello-world" is not correct.

Reply:
Ok, I see your point now. I didn't pay attention that `erratic` doesn't support EDNS.
Lets keep your changes for now. We may want to revise logic of `SizeAndDo()` in a separate PR if needed